src/services/email-service.ts

```diff
@@ -1,12 +1,210 @@
+
+import models = require('../models/models');
 import mailer = require('nodemailer');
+import config = require('../config');
+import Promise = require('bluebird');
 
+/**
+ * The email service which provides methods to send such things as activation
+ * emails.
+ */
 export class EmailService {
 
-  sendActivation(user: User): boolean {
-    // TODO: return promise, to implement
-    // Ian will do this
-   return false;
+  /**
+   * All email configuration options are stored here.
+   * @type {EmailConfigOptions}
+   */
+  private _transportConfig: EmailConfigOptions;
+
+  /**
+   * Initializes the email service instance.
+   */
+  constructor() {
+    this._transportConfig = config.Config.email;
   }
 
+  /**
+   * Returns the current transportation configuration options.
+   * @return {EmailConfigOptions} The email configuration options.
+   */
+  get transportConfig(): EmailConfigOptions {
+    return this._transportConfig;
+  }
+
+  /**
+   * Sets the transportation configuration options.
+   * @param  {EmailConfigOptions} config The email configuration options.
+   */
+  set transportConfig(config: EmailConfigOptions) {
+    this._transportConfig = config;
+  }
+
+  /**
+   * Sends the activation email to the specified user.
+   *
+   * @param  {models.User}                         user           The user.
+   * @param  {string}                              activationCode Their activation code.
+   * @return {Promise<nodemailer.SentMessageInfo>}                A Promise.
+   */
+  public sendActivation(user: models.User, activationCode: string): Promise<nodemailer.SentMessageInfo> {
+
+    var transporter = this.buildTransporter();
+
+
+    var mailOptions = {
+      from: config.Config.email.name + ' <' + config.Config.email.auth.user + '>',
+      to: user.firstName + ' ' + user.lastName + ' <' + user.email + '>',
+      subject: 'Activate Your Carpooling Account',
+      text: 'Hello ' + user.firstName + ', ' + "\r\n"
+            + 'You need to activate your account here: ' + config.Config.app.baseurl + '/activate?code=' + activationCode
+    };
+
+    return new Promise<nodemailer.SentMessageInfo>((resolve, reject) => {
+
+      transporter.sendMail(mailOptions, (error, sent) => {
+        if (error)
+        {
+          reject(error);
```

Comment:
Maybe return a custom error here? I would add a `ActivationCodeSendException` class in `errors.ts` file and some how wrap the message in `error` returned by `transporter.sendMail` in the the `ActivationCodeSendException`.

So you would call `reject` like following: `reject(new errors.ActivationCodeSendException(error.message))`. 

The reason for this is it would be very easy to catch an error that we know about than the generic error returned by `nodemailer`. So somewhere else we can do something like this:

```ts
emailService.sendActivation(....)
   .then(handleSucess)
   .catch(errors.ActivationCodeSendException, handleError)
```